lib/Doctrine/DBAL/Platforms/OraclePlatform.php

```diff
@@ -610,31 +646,80 @@ private function getAutoincrementIdentifierName(Identifier $table)
      */
     public function getListTableForeignKeysSQL($table)
     {
-        $table = $this->normalizeIdentifier($table);
-        $table = $this->quoteStringLiteral($table->getName());
+        return $this->getListForeignKeysSQL($table);
+    }
+
+    /**
+     * Returns the list of foreign keys for the current database.
+     *
+     * @param string $table
+     * @param string $database
+     *
+     * @return string
+     */
+    public function getListForeignKeysSQL($table = null, $database = null)
```

Comment:
This method needs to be called by OracleSchemaManager, hence the need to have it public. Are you suggesting to increase the public API to add  `getAllSomething()` methods? I'd be fine with that, we would have to add those to the `AbstractPlatform` class though... need @Ocramius input here I suppose.